common/remote/sockfile.cpp

```diff
@@ -2315,6 +2315,21 @@ class CRemoteFileIO : implements IFileIO, public CInterface
         // then also send sMode, cFlags
         unsigned short sMode = parent->getShareMode();
         unsigned short cFlags = parent->getCreateFlags();
+        switch ((compatIFSHmode)_compatmode)
+        {
+            case compatIFSHnone:
+                sMode = IFSHnone;
+                break;
+            case compatIFSHread:
+                sMode = IFSHread;
+                break;
+            case compatIFSHwrite:
+                sMode = IFSHfull;
+                break;
+            case compatIFSHall:
+                sMode = IFSHfull;
+                break;
+        }
```

Comment:
Solutions are: code could call setShareMode() on client side after creating remote file so that share mode is set correctly early, or this change to adjust sMode to be correct, or do not overwrite share mode with sMode.